lib/static-files-router.js

```diff
@@ -0,0 +1,287 @@
+        // images
+        jpg: 'image/jpeg',
+        jpeg: 'image/jpeg',
+        png: 'image/png',
+        bmp: 'image/bmp',
+        gif: 'image/gif',
+        svg: 'image/svg+xml'
+    };
+
+    /**
+     * Splits a full file name to its name and extension.
+     * @param fullFileName {string} file name to be split e.g. foo.txt
+     * @return {{name: string, extension: string}} splited parts
+     */
+    function splitFileName(fullFileName) {
+        var index = fullFileName.lastIndexOf(".");
+        return {name: fullFileName.substr(0, index), extension: fullFileName.substr(index + 1)}
+    }
+
+    /**
+     * Returns a file object that represent the specified file.
+     * @param unitFullName {string} full name of the unit
+     * @param resourceType {string} resource type
+     * @param fileName {string} file name with extension
+     * @param lookUpTable {Object} lookup table
+     * @param configs {Object} configurations
+     * @return {Object|null} if file exists a File object, otherwise <code>null</code>
+     */
+    function getFile(unitFullName, resourceType, fileName, lookUpTable, configs) {
+        var unitModel = lookUpTable.units[unitFullName];
+        if (!unitModel) {
+            return null; // unit does not exists
+        }
+        var filePathRelativeToUnit = null;
+        if (resourceType == "less") {
+            filePathRelativeToUnit = configs.constants.DIRECTORY_APP_UNIT_PUBLIC + "/less/"
+                                     + splitFileName(fileName).name + ".less";
+        } else {
+            filePathRelativeToUnit = configs.constants.DIRECTORY_APP_UNIT_PUBLIC + "/"
+                                     + resourceType + "/" + fileName;
+        }
+        // search in this unit
+        var filePath = configs.constants.DIRECTORY_APP_UNITS + "/" + unitFullName
+                       + filePathRelativeToUnit;
+        var file = new File(filePath);
+        if (file.isExists() && !file.isDirectory()) {
+            return file; // this unit has the file
+        }
+        // search in parent unit
+        var parentUnitFullName = unitModel.definition["extends"];
+        if (!parentUnitFullName) {
+            return null; // not an extended unit, hence does not has a parent unit
+        }
+        filePath = configs.constants.DIRECTORY_APP_UNITS + "/" + parentUnitFullName
+                   + filePathRelativeToUnit;
+        file = new File(filePath);
+        if (file.isExists() && !file.isDirectory()) {
+            return file; // parent unit has the file
+        }
+
+        return null; // file does not exits
+    }
+
+    function isLastModifiedDateEquals(file, request) {
+        return String(file.getLastModified()) == request.getHeader("If-Modified-Since");
+    }
+
+    /**
+     * Returns the MIME type of the specified file.
+     * @param path {string} file path or name
+     * @return {string} MIME type of the specified file
+     */
+    function getMimeType(path) {
```

Comment:
Fixed in PR https://github.com/wso2/carbon-uuf/pull/2 (see commit https://github.com/sajithar/carbon-uuf/commit/cfa43b7b168f22f195248c99e6a231602f27d6ef)